js/components/matrix.js

```diff
@@ -0,0 +1,366 @@
+        this.setDefaultValue();
+      } else {
+        if (typeof $vm.value === 'string') {
+          $vm.value = JSON.parse($vm.value);
+        }
+
+        _.forOwn($vm.value, function(key, value) {
+          var row = _.find($vm.rowOptions, function(row) {
+            return (_.has(row, 'label') && _.has(row, 'id')) ? row.id === value : row.label === value;
+          });
+
+          var rowIndex = _.findIndex($vm.rowOptions, function(row) {
+            return (_.has(row, 'label') && _.has(row, 'id')) ? row.id === value || row.label === value : row.label === value;
+          });
+
+          var col = _.find($vm.columnOptions, function(col) {
+            return (_.has(col, 'label') && _.has(col, 'id')) ? col.id === key : col.label === key;
+          });
+
+          var colIndex = _.findIndex($vm.columnOptions, function(col) {
+            return (_.has(col, 'label') && _.has(col, 'id')) ? col.id === key || col.label === key : col.label === key;
+          });
+
+          // setTimeout using to load all HTML and then execute below piece of code.
+          setTimeout(function() {
+            if (row && col) {
+              $vm.clickHandler(row, col, rowIndex, colIndex);
+            } else if (rowIndex >= 0) {
+              $vm.clickHandler($vm.rowOptions[rowIndex], $vm.columnOptions[colIndex], rowIndex, colIndex);
+            }
+          });
+        });
+      }
+    },
+
+    /**
+     * Clears the value for each row
+     *
+     * @returns {undefined}
+     */
+    clearValue() {
+      if (!this.$refs.matrix) {
+        return;
+      }
+
+      var $vm = this;
+
+      _.forEach(this.rowOptions, function(row, rowIndex) {
+        _.forEach($vm.columnOptions, function(col, colIndex) {
+          var el = $vm.getOptionId(rowIndex, colIndex, 'input');
+
+          $('#' + el).prop('checked', false);
+        });
+      });
+
+      if (this.value === '') {
+        this.setDefaultValue();
+      }
+    },
+
+    /**
+     * Check if all rows has column values or not
+     *
+     * @param {object} val - value of selected options
+     *
+     * @returns {undefined}
+     */
+    checkValue: function(val) {
+      if (typeof val === 'string' && val !== '') {
+        val = JSON.parse(this.value);
+      }
+
+      var checkFlag = '';
+      var $vm = this;
+      var columnOpt = [];
+
+      if (val === '') {
+        val = {};
+        checkFlag = 'clear';
+      } else if (!_.isEmpty(val)) {
+        var result = [];
+
+        _.forOwn(val, function(value) {
+          if (typeof value !== 'undefined') {
+            result.push(value);
+          }
+        });
+
+        if (result.length > 0) {
+          _.forEach(result, function(col) {
+            _.find($vm.columnOptions, function(column) {
+              if (column.label === col || column.id === col) {
+                columnOpt.push(column);
+              }
+            });
+          });
+        }
+
+        if (result.length === 0) {
```

Comment:
re-write this using && and same for set too

Reply:
done